### ISSUE TITLE
Add support for BlockEntityTag

### ISSUE THREAD

**Issue by ZephireNZ** [GlowstoneMC/Glowstone-Legacy#326]
Minecraft 1.8 now supports adding BlockEntityTag to the list of custom tags. This will cause these tags to be merged into a block's data once placed.

This tag should be supported on a broad level (ie in GlowBlock), so that any block placed will inherit these tags.

Relates to #325

**Comment by turt2live** [comment]
Doesn't really relate to #325 at all. More #119

**Comment by ZephireNZ** [comment]
Fair call. Regardless, I'll work on it has part of my work for banners (as they make prime use of it).

**Comment by turt2live** [comment]
Alright. (I also mis-linked banners, so there's that...)

**Comment by dequis** [comment]
What's this BlockEntityTag thing? Google tells me it's more bullshit related to the official command block plugin api™, but what does this mean for us?

**Comment by ZephireNZ** [comment]
It's an NBT tag added to the item. When a block is placed, they merge in any of the data in that tag to the block's NBT data.

**Comment by turt2live** [comment]
Ideally...
1. Add something to tag on item (dirt)
2. Use item to place block (now dirt is in the world)
3. Attempt to break block (plugin can check for that tag added at the item level)
4. Block broken. Item picked up still has that tag.

**Comment by ZephireNZ** [comment]
No, the tag is used for other stuff as well, such as setting the inventory of a block. Only way to get that NBT data back is to Ctrl+Middle Click. Otherwise it follows the normal block break stuff.

**Comment by turt2live** [comment]
It was a sample use case, not one that is perfectly valid :P

There's much more to it than what has been explained thus far by anyone. Ideally people should actually go read the wiki page about them.

**Comment by ZephireNZ** [comment]
Thoughts on how this could be implemented at the API level: 
- Add methods for getting a BlockState from the item. 
- Backed by the item's BlockEntityTag rather than a block's root NBT.
- Would get/set values just as it would with the in-game block.

**Comment by SpaceManiac** [comment]
If `GlowMetaItem` stored, persisted, and made available the appropriate CompoundTag, it would be fairly easy to have the tile entity in question load from the stack's tag with `TileEntity#loadNbt` after placement has occurred.

**Comment by ZephireNZ** [comment]
What are your thoughts on my idea of having GlowMetaItem return a BlockState for its NBT?

**Comment by ZephireNZ** [comment]
The only problem I can forsee is that the partial data may not have the full data needed for BlockState to work properly, leading to some interesting bugs. Maybe have a 'default' NBT for a given block, to ensure that the data is there (even if it's not on the actual tag)

**Comment by SpaceManiac** [comment]
Not sure, it seems like over-complicating it (especially since BlockState is tied to a location). The most sane solution seems to me to be that after the block is placed, if it is both a tile-entity and the item in hand has the extra tag, load data from the extra tag.

**Comment by ZephireNZ** [comment]
My main reasoning for adding the BlockState is so you can easily access it from the API. Otherwise, as I've seen with Banners, the ItemMeta become almost an exact replica of the BlockState, which seems very wasteful.
